File: Easy/longest_common_prefix.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        if len(strs) == 0 or len(strs) > 200:
            return ""
        
        def sortByLen(s):
            return len(s)
        
        strs.sort(key=sortByLen)
        lenShortText = len(strs[0])
        collectPrefix = []
        
        for i in range(lenShortText):
            if len(collectPrefix) == i:
                collectPrefix.append(strs[0][i])
                
            if i > len(collectPrefix) or len(collectPrefix) == 0:
                break
                
            for _, text in enumerate(strs):               
                if not text:
                    continue 
                
                if text[i] != collectPrefix[i]:
                    if len(collectPrefix) > 0:
                        collectPrefix.pop()
                    break        
        
        return "".join(collectPrefix)
    
print(Solution().longestCommonPrefix(["flower","flow","flight"])) #fl
print(Solution().longestCommonPrefix(["flower","flow",""])) #fl
print(Solution().longestCommonPrefix(["flower","flower","flower","flower"])) #flower
print(Solution().longestCommonPrefix(["ab", "a"])) #a
print(Solution().longestCommonPrefix(["", ""])) #
print(Solution().longestCommonPrefix(["a"])) #a

File: Easy/longest_common_prefix.php
<?php
class Solution
{

    /**
     * @param String[] $strs
     * @return String
     */
    function longestCommonPrefix($strs)
    {
        if (count($strs) == 0 || count($strs) > 200) {
            return "";
        }

        usort($strs, function ($a, $b) {
            return strlen($a) - strlen($b);
        });

        $lenShortText = strlen($strs[0]);
        $collectPrefix = array();
        
        for ($i = 0; $i < $lenShortText; $i++) { 
            if (count($collectPrefix) == $i) {
                array_push($collectPrefix, $strs[0][$i]);
            }

            if ($i > count($collectPrefix) || count($collectPrefix) == 0) {
                break;
            }

            foreach($strs as $text) {
                if ($text === "") {
                    continue;
                }
                
                if ($text[$i] != $collectPrefix[$i]) {
                    array_pop($collectPrefix);
                    break;
                }
            }
        }

        return implode("", $collectPrefix);
    }
}

$solution = new Solution;
print_r($solution->longestCommonPrefix(array("flower", "flow", "flight")) . "\n"); //fl
print_r($solution->longestCommonPrefix(array("flower", "flow", "")) . "\n"); //
print_r($solution->longestCommonPrefix(array("flower", "flower", "flower", "flower")) . "\n"); //flower
print_r($solution->longestCommonPrefix(array("ab", "a")) . "\n"); //a
print_r($solution->longestCommonPrefix(array("", "")) . "\n"); //
print_r($solution->longestCommonPrefix(array("a")) . "\n"); //a